SSA/Core/Util/ConcreteOrMVar.lean

```diff
@@ -48,5 +48,22 @@ def instantiate (as : Vector α φ) : ConcreteOrMVar α φ → α
   | .concrete w => w
   | .mvar i => as.get i
 
+@[simp]
+def instantiate_concrete_eq (as : Vector α φ) :
+  (ConcreteOrMVar.concrete w).instantiate as = w := by rfl
+
+@[simp]
+lemma instantiate_mvar_zero {hφ : List.length (w :: ws) = φ} {h0 : 0 < φ} :
+    ConcreteOrMVar.instantiate (Subtype.mk (w :: ws) hφ)  (ConcreteOrMVar.mvar ⟨0, h0⟩) = w := by
+  simp [instantiate]
+  simp [Vector.get]
+
+@[simp]
+lemma instantiate_mvar_zero' :
+    (mvar (φ := 1) ⟨0, by simp⟩).instantiate (Subtype.mk [w] (by simp)) = w := rfl
+
+@[simp]
+lemma instantiate_mvar_zero'' :
+    (mvar (φ := 1) 0).instantiate (Subtype.mk [w] h1) = w := rfl
```

Comment:
The difference between these three lemmas is not very clear to me, and obfuscated by different spellings of identical concepts (i.e., `ConcreterOrMVar.instantiate` vs dot-notation `(...).instantiate` and `Subtype.mk` vs magic braces `⟨_, _⟩`). I'd like to see this be more uniform, and I'd like to see a docstring to explain the differences